### PR TITLE
ARROW-6875: [FlightRPC] implement criteria for ListFlights

### PR DESCRIPTION
**Pull request by lidavidm**
I'm now working on the integration tests. I got to the end and realized it would be an enormous PR, so instead I'm splitting out the various fixes I made. 

None of the languages fully supported passing filter criteria in ListFlights.

### CODE REVIEW

File: python/pyarrow/tests/test_flight.py [hunk 1]
```diff
@@ -129,6 +129,15 @@ def __init__(self, location=None, **kwargs):
             b'dicts': simple_dicts_table,
         }
 
+    def list_flights(self, context, criteria):
+        if criteria:
```
Comment (pitrou) [inline python/pyarrow/tests/test_flight.py]: Should you check a well-defined criteria value here to make the test more interesting?

**Review by pitrou — APPROVED** [review]
LGTM, just a question.

File: python/pyarrow/tests/test_flight.py (hunk 1 above)
Comment (lidavidm) [reply python/pyarrow/tests/test_flight.py]: Good idea, I've updated the test.

**Comment by pitrou** [comment]
Thank you @lidavidm !

**Comment by lidavidm** [comment]
Thank you for the review :)

**Pull request closed without merging**